views/js/runner/plugins/security/preventCopy.js

```diff
@@ -140,19 +140,55 @@ define([
             var testRunner = this.getTestRunner();
             var prohibitedKeyFunc;
             var prohibitedKeyDebounce;
-
-            function prevent(event) {
-                if ($(event.target).closest('textarea, input, [contenteditable]')) {
+            var isIe = typeof window.clipboardData !== 'undefined';
+
+            function replaceSelection(target, newValue) {
+                var oldValue = target.value.toString().substring(target.selectionStart, target.selectionEnd);
+                var caretPosition = target.selectionStart;
+                var p1 = target.value.substring(0,caretPosition);
+                var p2 = target.value.substring(caretPosition+oldValue.length,target.value.length);
+                var pa = newValue;  //new item
+                target.value = p1 + pa + p2;
+                target.selectionStart = caretPosition + newValue.length;
+                target.selectionEnd = caretPosition + newValue.length;
+                target.focus();
+            }
+            function onCopyCut(event) {
+                var target = $(event.target).closest('textarea, input, [contenteditable]')[0];
+                var text = target.value.toString().substring(target.selectionStart, target.selectionEnd);
+                if (target) {
+                    if (isIe) {
+                        window.clipboardData.setData('Text', '');
+                    } else {
+                        event.clipboardData.setData('text/plain', '');
+                    }
+                    event.preventDefault();
+                    if (event.type === 'cut') {
+                        replaceSelection(target, '');
+                    }
+                    $(target).attr('data-clipboard', text);
+                    return;
```

Comment:
It would be better to rely on an `else` block instead. This will make the code clearer. Having an early `return` in the code should be avoided when it is not needed.